### DISCUSSION TITLE
is there a way to auto delete old audit records?

### DISCUSSION THREAD

**Discussion by rebbieboi** [owen-it/laravel-auditing#897] [Q&A] [answered]
Is there a built-in tool to auto-delete old audit records? or do I just have to create a scheduler to do it for me?

Thank you!

**Comment by parallels999** [comment, marked as answer]
https://github.com/owen-it/laravel-auditing/blob/42ebec8a5abfcc5d61b563eb95c77dd597be4185/config/audit.php#L121-L130
[https://laravel-auditing.com/guide/auditable-configuration.htm](https://laravel-auditing.com/guide/auditable-configuration.html#number-of-audit-records)

**Reply by rebbieboi** [reply]
will that overwrite old records to maintain the defined threshold OR will it stop creating new records when the threshold is reached?

**Reply by rebbieboi** [reply]
that is what I'm trying to know as well. because both can be possible if you wanted it to be and since there's no further description in there I won't really know

**Reply by rebbieboi** [reply]
> Specify a threshold for the amount of Audit records a model can have.  Zero means no limit.

I missed the part where it answered my question.

For those interested in getting the answer instead of hearing a non answer. **It will overwrite the old records.**

**Reply by angeljqv** [reply]
[https://laravel-auditing.com/guide](https://cdn.someecards.com/someecards/usercards/people-like-you-are-the-reason-shampoo-has-directions-f70da.png)

**Reply by justinmoh** [reply]
Revisiting an old app developed a few years ago and concerning about this issue as well. Actually this part of the config is what I find somewhat not intuitive. 

My instinct tells me that we must have a way to "Add limit to the total records we're keeping", it's such a common requirement at most use case. 

But instead this config is about "Add limit to each model", not what I wanted -- or never have thought of that.

I wonder if the team has any plan on improving this part. For the time being, I'm just using my quick and dirty method:

```php
class Audit extends \OwenIt\Auditing\Models\Audit
{
    public static function prune(int $daysToKeep = 90)
    {
        DB::table(config('audit.drivers.database.table'))
            ->where('created_at', '<', today()->subDays($daysToKeep))
            ->delete();
    }
}
```

I then use it in a command and schedule it monthly.

**Reply by parallels999** [reply]
https://github.com/owen-it/laravel-auditing/discussions/925
>I wonder if the team has any plan on improving this part

There are no plans in the near future.
